src/ensemble/basic_ensemble_solve.jl

```diff
@@ -53,51 +95,66 @@ end
 function __solve(prob::AbstractEnsembleProblem,
     alg::Union{AbstractDEAlgorithm, Nothing},
     ensemblealg::BasicEnsembleAlgorithm;
-    trajectories, batch_size = trajectories,
+    trajectories, batch_size = trajectories, progress_aggregate=true,
     pmap_batch_size = batch_size ÷ 100 > 0 ? batch_size ÷ 100 : 1, kwargs...)
-    num_batches = trajectories ÷ batch_size
-    num_batches < 1 &&
-        error("trajectories ÷ batch_size cannot be less than 1, got $num_batches")
-    num_batches * batch_size != trajectories && (num_batches += 1)
-
-    if num_batches == 1 && prob.reduction === DEFAULT_REDUCTION
-        elapsed_time = @elapsed u = solve_batch(prob, alg, ensemblealg, 1:trajectories,
-            pmap_batch_size; kwargs...)
-        _u = tighten_container_eltype(u)
-        return EnsembleSolution(_u, elapsed_time, true)
-    end
+    logger = progress_aggregate ? AggregateLogger(current_logger()) : current_logger()
```

Comment:
does this cause dynamic dispatch?

Reply:
Not any more than `current_logger` already does
```
julia> Base.return_types(current_logger, Tuple{})
1-element Vector{Any}:
 Base.CoreLogging.AbstractLogger

julia> test(cond) = cond ? ConsoleLogger() : current_logger()
test (generic function with 1 method)

julia> Base.return_types(test, Tuple{Bool})
1-element Vector{Any}:
 Base.CoreLogging.AbstractLogger
```

Reply:
Yes, which is why we don't use it.

Reply:
What I mean is, any logging call is always going to be dynamic dispatch because the logger is context dependent and can be changed at runtime